02_knight_and_princess/kirill_tolstobrov/main.cpp

```diff
@@ -0,0 +1,145 @@
+#include <fstream>
+#include <iostream>
+#include <iterator>
+#include <map>
+#include <utility>
+#include <vector>
+
+using VertexId = int;
+using EdgeId = int;
+
+class Vertex {
+ public:
+  explicit Vertex(const VertexId& init_id = 0) : id(init_id) {}
+  void add_edge(const EdgeId& edge_id) { connected_edges_.push_back(edge_id); }
+  operator std::string() {
+    std::string result = "";
+    result += "    {\n      \"id\": " + std::to_string(id);
+    result += ",\n      \"edge_ids\": [";
+
+    for (int i = 0; i < connected_edges_.size(); i++) {
+      result += std::to_string(connected_edges_[i]);
+      if (i == connected_edges_.size() - 1) {
+        result += "]\n";
+      } else {
+        result += ", ";
+      }
+    }
+    result += "\n    }";
+    return result;
+  }
+
+  const VertexId id;
+
+ private:
+  std::vector<EdgeId> connected_edges_;
+};
+
+class Edge {
+ public:
+  Edge(const EdgeId& init_id, const VertexId& v1_init, const VertexId& v2_init)
+      : id(init_id), vertex1_id(v1_init), vertex2_id(v2_init){};
+  operator std::string() {
+    std::string result = "";
+    result += "    {\n      \"id\": " + std::to_string(id);
+    result += ",\n      \"vertex_ids\": ";
+
+    result += "[" + std::to_string(vertex1_id);
+    result += ", " + std::to_string(vertex2_id);
+    result += "]\n    }";
+
+    return result;
+  }
+
+  const EdgeId id;
+  const VertexId vertex1_id;
+  const VertexId vertex2_id;
+};
+
+class Graph {
+ public:
+  int get_vertices_amount() const { return vertices_.size(); }
+  int get_edges_amount() const { return edges_.size(); }
+
+  void add_new_vertex() { vertices_.push_back(Vertex(vertices_.size())); }
```

Comment:
nit: Для конструировании нового объекта в коллекцию, есть специальный метод - `emplace`. Поэтому вместо создании объекта и его копирования, достаточно было бы написать так:
```suggestion
  void add_new_vertex() { vertices_.emplace_back(vertices_.size()); }
```